builtin/rebase.c

```diff
@@ -245,7 +247,22 @@ int cmd_rebase(int argc, const char **argv, const char *prefix)
 	if (!options.onto_name)
 		options.onto_name = options.upstream_name;
 	if (strstr(options.onto_name, "...")) {
-		die("TODO");
+		/* From here on more ugly codes */
+		if ((left == (strstr(options.onto_name, "%...*")) ||
```

Comment:
You should not need to parse out `left` and `right`... As there is `get_oid_mb()`. I'm online on IRC now, for an hour or so, if you want...

Reply:
Besides, if you want to parse `left` and `right`, it would be better to actually store the result of `strstr(options.onto_name, "...")`. The `%...*` construct really only makes sense in the context of Unix shell scripting: `${name%...*}` will take the value of the variable `name`, see whether it ends in three dots followed by an arbitrary string (or nothing) and strip that suffix if it exists, otherwise it will simply interpolate to the value of `name`.

To make use of the return value of `strstr(options.onto_name, "...")`, please remember that it returns the pointer to the first occurrence of `...` in `onto_name` (or `NULL`, if there is no `...` in there). Then, `dotdotdot - options.onto_name` (where I use `dotdotdot` as the tentative name of the variable to store the return value) will tell you the length of `left`, and it of course starts at `options.onto_name`. So something like `xstrndup(options.onto_name, dotdotdot - options.onto_name)` would give you the equivalent of the shell script's `left`.

But as I said, `get_oid_mb()` does all of that for you, no need to repeat that code.